lib/src/platform_sliver_app_bar.dart

```diff
@@ -0,0 +1,297 @@
+    return SliverAppBar(
+      //Common
+      key: data?.widgetKey ?? widgetKey,
+      leading: data?.leading ?? leading,
+      automaticallyImplyLeading:
+          data?.automaticallyImplyLeading ?? automaticallyImplyLeading ?? true,
+      backgroundColor: data?.backgroundColor ?? backgroundColor,
+      stretch: data?.stretch ?? stretch ?? false,
+      title: data?.title ?? title,
+
+      //Material only
+      actions: data?.actions,
+      flexibleSpace: data?.flexibleSpace,
+      bottom: data?.bottom,
+      elevation: data?.elevation,
+      shadowColor: data?.shadowColor,
+      forceElevated: data?.forceElevated ?? false,
+      foregroundColor: data?.foregroundColor,
+      iconTheme: data?.iconTheme,
+      actionsIconTheme: data?.actionsIconTheme,
+      primary: data?.primary ?? true,
+      centerTitle: data?.centerTitle,
+      excludeHeaderSemantics: data?.excludeHeaderSemantics ?? false,
+      titleSpacing: data?.titleSpacing,
+      collapsedHeight: data?.collapsedHeight,
+      expandedHeight: data?.expandedHeight,
+      floating: data?.floating ?? false,
+      pinned: data?.pinned ?? false,
+      snap: data?.snap ?? false,
+      stretchTriggerOffset: data?.stretchTriggerOffset ?? 100.0,
+      onStretchTrigger: data?.onStretchTrigger,
+      shape: data?.shape,
+      toolbarHeight: data?.toolbarHeight ?? kToolbarHeight,
+      leadingWidth: data?.leadingWidth,
+      toolbarTextStyle: data?.toolbarTextStyle,
+      titleTextStyle: data?.titleTextStyle,
+      systemOverlayStyle: data?.systemOverlayStyle,
+      forceMaterialTransparency: data?.forceMaterialTransparency ?? false,
+      clipBehavior: data?.clipBehavior,
+    );
+  }
+
+  @override
+  CupertinoSliverNavigationBar createCupertinoWidget(BuildContext context) {
+    final data = cupertino?.call(context, platform(context));
+
+    return CupertinoSliverNavigationBar(
+      //Common
+      key: data?.widgetKey ?? widgetKey,
+      leading: data?.leading ?? leading,
+      automaticallyImplyLeading:
+          data?.automaticallyImplyLeading ?? automaticallyImplyLeading ?? true,
+      backgroundColor: data?.backgroundColor ?? backgroundColor,
+      stretch: data?.stretch ?? stretch ?? false,
+      largeTitle: data?.title ?? title,
+
+      //Cupertino only
+      automaticallyImplyTitle: data?.automaticallyImplyTitle ?? true,
+      alwaysShowMiddle: data?.alwaysShowMiddle ?? true,
+      previousPageTitle: data?.previousPageTitle,
+      middle: data?.middle,
+      trailing: data?.trailing,
+      border: data?.border ?? _kDefaultNavBarBorder,
+      brightness: data?.brightness,
+      padding: data?.padding,
+      transitionBetweenRoutes: data?.transitionBetweenRoutes ?? true,
+      heroTag: data?.heroTag ?? _defaultHeroTag,
+    );
+  }
+}
+
+//! Copied from file: /opt/homebrew/Caskroom/flutter/3.10.0/flutter/packages/flutter/lib/src/cupertino/nav_bar.dart
+const Color _kDefaultNavBarBorderColor = Color(0x4D000000);
```

Comment:
Regarding these local variables, I didn't find a better solution, so I had to copy and paste them here intact.